internal/engine/wazevo/engine.go

```diff
@@ -227,27 +227,83 @@ func (e *engine) compileModule(ctx context.Context, module *wasm.Module, listene
 	needCallTrampoline := callTrampolineIslandSize > 0
 	var callTrampolineIslandOffsets []int // Holds the offsets of trampoline islands.
 
-	for i := range module.CodeSection {
-		if wazevoapi.DeterministicCompilationVerifierEnabled {
-			i = wazevoapi.DeterministicCompilationVerifierGetRandomizedLocalFunctionIndex(ctx, i)
-		}
+	type CompiledLocalFuncResult struct {
+		Body             []byte
+		RelsPerFunc      []backend.RelocationInfo
+		IDX              wasm.Index
+		SourceOffsetInfo []backend.SourceOffsetInfo
+	}
+
+	compiledFuncs := make([]CompiledLocalFuncResult, len(module.CodeSection))
+
+	workers := experimental.GetCompilationWorkers(ctx)
+
+	wg := sync.WaitGroup{}
+	wg.Add(workers)
+
+	ctx, cancel := context.WithCancelCause(ctx)
+	defer cancel(nil)
```

Comment:
i don't think we need this. compilations are not in-fact cancelable (we don't have any context checking in the compilation routines)